ethcontract-common/src/artifact/hardhat.rs

```diff
@@ -668,4 +848,109 @@ mod test {
         assert_eq!(a.networks.len(), 1);
         assert_eq!(a.networks["4"].address, address(0xBA));
     }
+
+    #[test]
+    fn load_multi_allow_contract_name() {
+        let artifact = HardHatLoader::new()
+            .allow_contract("A")
+            .load_from_str(Format::MultiExport, MULTI_EXPORT)
+            .unwrap();
+
+        assert_eq!(artifact.len(), 1);
+
+        let a = artifact.get("A").unwrap();
+        assert_eq!(a.name, "A");
+        assert_eq!(a.networks.len(), 2);
+        assert_eq!(a.networks["1"].address, address(0xA));
+        assert_eq!(a.networks["4"].address, address(0xAA));
+
+        let artifact = HardHatLoader::new()
+            .allow_contract("X")
+            .load_from_str(Format::MultiExport, MULTI_EXPORT)
+            .unwrap();
+
+        assert_eq!(artifact.len(), 0);
+    }
+
+    #[test]
+    fn load_multi_deny_contract_name() {
+        let artifact = HardHatLoader::new()
+            .deny_contract("A")
+            .load_from_str(Format::MultiExport, MULTI_EXPORT)
+            .unwrap();
+
+        assert_eq!(artifact.len(), 1);
```

Comment:
Perhaps I misunderstood the meaning of denied contracts but why is there artifact length now 1? All I see at this point is a new hardhat loader with a denied contract.

Reply:
We've created a loader with denied contract, then loaded an artifact. Artifact JSON contained two contract: `A` and `B`. Since we've denied loading `A`, the only contract that was loaded into the artifact is `B`, hence `artifact.len() == 1`.

Reply:
It `.load_from_str(Format::MultiExport, MULTI_EXPORT)`, where `MULTI_EXPORT` is a Hardhat multi-contract export with two contracts `A` and `B`. Since `A` was denied, you would expect the loaded artifact to only have a single contract `B`.

Reply:
I think the json contains a contract A and a contract B. In the previous test we use an allow list so only A gets loaded and in this test we use a deny list so only B gets loaded.